src/main/java/com/mewna/catnip/extension/Extension.java

```diff
@@ -154,61 +126,14 @@ default Extension injectOptions(@Nonnull final Function<CatnipOptions, CatnipOpt
     }
     
     /**
-     * Add a consumer for the specified event type.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(EventType)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type The type of event to listen on.
-     * @param <T>  The object type of event being listened on.
-     *
-     * @return The vert.x message consumer.
-     */
-    <T> MessageConsumer<T> on(@Nonnull EventType<T> type);
-    
-    /**
-     * Add a consumer for the specified event type with the given handler
-     * callback.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(EventType, Consumer)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type    The type of event to listen on.
-     * @param handler The handler for the event object.
-     * @param <T>     The object type of event being listened on.
-     *
-     * @return The vert.x message consumer.
+     * Callback called once the Extension has been loaded by the {@link ExtensionManager}.
      */
-    <T> MessageConsumer<T> on(@Nonnull EventType<T> type, @Nonnull Consumer<T> handler);
-    
-    /**
-     * Add a consumer for the specified event type.
-     * <p>
-     * This method behaves similarly to {@link Catnip#on(DoubleEventType)},
-     * but the event listeners are unregistered when the extension unloads
-     *
-     * @param type The type of event to listen on.
-     * @param <T>  The first type of event being listened on.
-     * @param <E>  The second type of event being listened on.
-     *
-     * @return The vert.x message consumer.
-     */
-    <T, E> MessageConsumer<Pair<T, E>> on(@Nonnull DoubleEventType<T, E> type);
+    default void onLoaded() {
```

Comment:
Should add documentation that explains that it's run on the Rx scheduler and not blocking the calling thread

Reply:
it's now synchronous owo